client/logic/game/cursor.gd

```diff
@@ -0,0 +1,9 @@
+class_name Cursor
+extends Marker2D
+
+func _process(delta: float) -> void:
+	position = get_global_mouse_position()
+
+func get_tile_position() -> Vector2i:
+	var grid_size: int = ProjectSettings.get_setting("global/grid_size").y
```

Comment:
I'm not a big fan of this, I understand why you are using it, but...

Reply:
Exactly

Reply:
ok, we had the discussion already, I will not raise it again. Fine for me.